keps/sig-node/20190226-pod-overhead.md

```diff
@@ -98,8 +121,7 @@ Introduce a Pod.Spec.Resources field on the pod to specify the pods overhead.
 ```
 Pod {
   Spec PodSpec {
-    // Overhead is the resource overhead consumed by the Pod, not including
-    // container resource usage. Users should leave this field unset.
+    // Overhead is the resource overhead incurred from the runtime.
     // +optional
     Overhead *ResourceRequirements
```

Comment:
A comment from @tallclair on initial PR:
```
I don't want to fall too much into the YAGNI trap, but too extensions I could see wanting in the future are:

Percentage-based overhead (i.e. pod overhead scales with size of pod)
Per-container overhead
If we were adding these features now, I might suggest an API roughly like this:

Overhead *Overhead
...
type Overhead struct {
  PodFixed *ResourceRequirements
  PodScaled *ScaledResourceRequirements
  ContainerFixed *ResourceRequirements
  ContainerScaled *ScaledResourceRequirements
}
If we want to leave the door open for this, we might want to make the initial API resemble this structure, i.e.:

type Overhead struct {
  PodFixed *ResourceRequirements  // Or just Pod *ResourceRequirements
}
WDYT?
```

Reply:
I like the % based concept, but need to think more on this before commenting on the suggested API.  Based on this initial reaction, I tend to think the "door open" may be a good first step.

/cc @mcastelino

Reply:
i am not familiar enough with situations where overhead would be a scalar rather than a fixed cost.  if we think there will be different applications of overhead, then having "PodFixed" from the start is probably safest.

Reply:
I went ahead and updated to just use the Overhead structure as Tim proposed.  I am not ready to embrace the expanded fields until we can gather more details; PodFixes is a very good start.